external/js/cothority/src/network/proto.ts

```diff
@@ -251,15 +218,28 @@ export class ServerIdentity extends Message<ServerIdentity> {
     }
 
     /**
-     * Returns websocket version of this.url if set, otherwise converts the server
-     * address to match the websocket format.
+     * Returns websocket URL to connect to this ServerIdentity.
      * @returns the websocket address
      */
     getWebSocketAddress(): string {
-        if (this.url) {
-            return ServerIdentity.urlToWebsocket(this.url);
+        if (this.url !== undefined && this.url !== "") {
```

Comment:
```javascript
if (this.url) {
```

Reply:
Why so? It's less precise and wrong if `this.url == null` (as `tsconfig.strictNullChecks` is not set).

Reply:
because if `this.url` is equal to null, the if clause will be true

Reply:
Yup, because it shouldn't be `null` and thus indicating a bug. (it's also because denoting missing value by empty string wrong IMO, so expliciting the fact that we support empty string makes it more readable)

Reply:
It's not a bug to have either `undefined`, `null` or an empty string. The type allows it so it needs to be handled.

Reply:
It's using raw types as the class is a message coming from the network.